### ISSUE TITLE
authn-service fails due to A1 data collector token projects error

### ISSUE THREAD

**Issue by bcmdarroch** [chef/automate#2955]
## Describe the bug
[Reported by `@axelrtgs` in the community slack](https://chefcommunity.slack.com/archives/CAZSL536K/p1582575944004700)

On updating to the latest Automate: 
```sudo chef-automate status
Status from deployment with channel [current] and type [local]
Service Name            Process State  Health Check  Uptime (s) PID  
deployment-service      running        ok            10060      29512
backup-gateway          running        ok            94         12640
automate-postgresql     running        ok            93         12684
automate-pg-gateway     running        ok            92         12767
automate-elasticsearch  running        ok            91         12810
automate-es-gateway     running        ok            91         12784
automate-ui             running        ok            91         12796
pg-sidecar-service      running        ok            91         12888
cereal-service          running        ok            90         12946
event-service           running        ok            90         13043
es-sidecar-service      running        ok            72         14896
event-feed-service      running        ok            71         14913
authz-service           running        ok            88         13175
automate-dex            running        ok            88         13298
teams-service           running        ok            87         13336
authn-service           DOWN           unknown       1          0    
secrets-service         running        ok            86         13511
applications-service    running        ok            85         13544
notifications-service   running        ok            84         13669
nodemanager-service     running        ok            85         13636
compliance-service      running        CRITICAL      18         16255
license-control-service running        ok            83         13959
local-user-service      running        ok            82         14143
session-service         running        ok            82         14090
ingest-service          running        ok            73         14839
infra-proxy-service     running        ok            82         14203
config-mgmt-service     running        ok            71         14935
data-feed-service       running        ok            80         14324
event-gateway           running        ok            79         14516
automate-gateway        running        unknown       79         14548
automate-load-balancer  running        ok            46         15584
12:26
```
According to `@axelrtgs`:
"the symptom of this is after login it can no longer validate our license and all our clients are reporting a 403 sending data to the datacollector"

## To Reproduce
The issue only shows up when there is an `a1_data_collector_token` configured.
```
rm -rf results/*.hart
hab studio enter
start_all_services
echo '[auth_n]
  [auth_n.v1]
    [auth_n.v1.sys]
      [auth_n.v1.sys.service]
        a1_data_collector_token = "LEGACY_TOKEN_HERE"' >> a1.toml
chef-automate config patch a1.toml
sl | grep authn-service
```
Logs should show this error: `authn-service.default(O): failed to initialize server: could not populate the legacy data collector token: rpc error: code = PermissionDenied desc = Projects cannot be modified because the [] subject is missing iam:projects:assign on this project: ["(unassigned)"]`

## Expected behavior
Configuring an a1 data collector token should work regardless of IAM version.

## Versions
- Automate Build Number: 20200220011437

**Comment by susanev** [comment]
Some more information
- folks were on iam v2
- they use auto-upgrade
- they recently deleted all their legacy policies
    - they had no issues after doing that
    - until the upgrade which caused service restarts

**Comment by sdelano** [comment]
This issue was fixed by #2961 

The conditions by which this were triggered are as follows:

1. Automate `authn-service` can be configured with a pre-defined token to support seamless upgrades from Automate 1. If this is configured, the service will try to create the token at startup if it does not exist.
2. The creation half of the create-token-at-startup mechanism has been broken on IAMv2 since October (https://github.com/chef/automate/commit/a2e1a87bc749cd89d5011c82b5379d6fdf8fdc7f)
3. This resulted in failures in the following scenarios
  a. A user using IAMv2 tries to configure a legacy data collector token (breaks immediately)
  b. A user using IAMv2 that had previously configured and successfully created a legacy data collector token deletes that token, triggering the creation logic on service restart. Service restart could be any time, but likely happens during an upgrade.
